receiver/dotnetdiagnosticsreceiver/go.mod

```diff
@@ -1,49 +0,0 @@
-module github.com/open-telemetry/opentelemetry-collector-contrib/receiver/dotnetdiagnosticsreceiver
```

Comment:
> Looks like we also didn't add a deprecated/unmaintained log warning. Should that be a requirement prior to removal?

Should this change be reverted as well then? https://github.com/open-telemetry/opentelemetry-collector-releases/commit/f8adde58bc1db8d467785d4fe7f21986de5599cd

Reply:
@dmitryax @mx-psi thoughts? ^^^

Reply:
That sounds great to me. I'd still suggest adding a warning for at least a couple of releases before dropping the component, just in case other custom builds use it.

Reply:
@mx-psi @dmitryax PTAL at https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20740